web/src/compose.js

```diff
@@ -169,7 +178,137 @@ export function send_message_success(request, data) {
     }
 }
 
-export function send_message(request = create_message_object()) {
+let split_long_messages_banner_shown = false;
+
+export function update_split_long_messages_info_banner() {
+    if (check_need_to_split_long_messages()) {
+        show_split_long_messages_info_banner();
+        compose_ui.autosize_textarea($("textarea#compose-textarea"));
+    } else {
+        clear_split_long_messages_info_banner();
+    }
+}
+
+const SPLIT_DELIMITER = "\n\n\n";
+/**
+ * checks whether the message actually needs to be split, as it is possible that `split_long_messages` is enabled but the message doesn't contain the delimiter.
+ */
```

Comment:
Please line-wrap your comments; every editor has a feature to do it and it's a lot nicer.